Add abstract FileLoader for Network operations

tamimattafi opened the pull request:
Firstly, thank you for this amazing library! I've been using it for as long as I used Compose Multiplatform. However, I needed to load some animations using a URL and faced some issues.

## Problem
Loading animations from a URL fails on Android and JVM if the app depends on Ktor 3.0.0 (Didn't test on other platforms)
Here's the error:
```kotlin
java.lang.NoClassDefFoundError: Failed resolution of: Lio/ktor/client/plugins/HttpRequestRetry;
```

Passing a custom client (Created using Ktor 3.0.0) throws another error:
```kotlin
java.lang.NoSuchMethodError: No static method toByteArray(Lio/ktor/utils/io/ByteReadChannel;Lkotlin/coroutines/Continuation;)Ljava/lang/Object; in class Lio/ktor/util/ByteChannelsKt; or its super classes (declaration of 'io.ktor.util.ByteChannelsKt' appears in /data/data/com.app.example/code_cache/.overlay/base.apk/classes21.dex)
```

## Solution
For the sake of simplicity, I copied `compottie-network` module to `compottie-network-ktor3` and adjusted code in two places:
1. Change dependencies of the new module to use Ktor 3.0.0-beta02 and Kotlin.io's latest version
2. Used `body<Source>()` to read bytes from response instead of channel (Using kotlin.io)

Some classes are left without any changes and could be moved to something like `compottie-network-core`, to eliminate duplication. However, I decided not to change existing code until a request from the reviewer.

Commit:
fix: create new module for ktor3 to fix loading crashes

alexzhirkevich commented:
Thanks. The intention is good but creating multiple modules for different Ktor versions is not the right way. There should be a core network module with abstract http client without ktor dependency, i'll do it later

Commit:
feat: add FileLoader abstraction, remove ktor3 module

tamimattafi commented:
Hello @alexzhirkevich! Thank you for your feedback.

I agree that adding multiple modules is not the correct way, I got the idea from `coil` since it's having many changes and in order not to break older users, they add new modules.

However, I made some changes to the PR:
1. Removed the extra module (`compottie-network-ktor3`)
2. Added `FileLoader` (interface)
2. Added some functions with `FileLoader`, but kept all old public functions and classes without change, this way, we can pass an abstract `FileLoader` but still don't break old versions that pass a custom `HttpClient` (No breaking changes introduced)

Could you please review these changes?
In the future, we can split this module and extract Ktor, but this will require a major update.

I tested the sample app, no changes had to be made there, and it works well on the platforms I tested on (Android, Desktop on macOS, and JS on Chrome).
Also, I tested the new changes in my project with a custom `FileLoader` that uses Ktor 3.0.0 instead of the default implementation, on Android and iOS, and it works well.

Commit:
refactor: revert readme and lib versions

Commit:
fix: add equals and hashCode to KtorFileLoader

Commit:
docs: add javadoc for newly added methods and classes

Commit:
Merge branch 'standalone-main' into fix/ktor3_crash
